docker/build_scripts/python-tag-abi-tag.py

```diff
@@ -2,9 +2,16 @@
 # See PEP 425 for exactly what these are, but an example would be:
 #   cp27-cp27mu
 
-from packaging.tags import sys_tags
+from packaging.tags import sys_tags, Tag
+from wheel.bdist_wheel import get_abi_tag
 
 
 # first tag is always the more specific tag
 tag = next(sys_tags())
+
+# there are some inconsistencies so check against wheel
+wheel_abi_tag = get_abi_tag()
+if tag.abi != wheel_abi_tag:
+    assert tag.interpreter.startswith("graalpy")
+    tag = Tag(tag.interpreter, wheel_abi_tag, tag.platform)
```

Comment:
@timfel,

It seems that packaging does not agree with wheel w.r.t. to the ABI tag. This is the sign that something bad will happen in the future (I guess when pip will update packaging: https://github.com/pypa/pip/issues/11715).
The difference comes from https://github.com/pypa/packaging/commit/30554f503a5368aabf6ee571bd9cdf29d5f00c12

Unfortunately, the ecosystem did not move forward with a sysconfig like that to express this ABI. xref https://discuss.python.org/t/sysconfig-should-provide-more-information-to-help-packaging/22950 and https://github.com/pypa/packaging/issues/330

Reply:
We have a PR to wheel to fix the `get_abi_tag` function in `src/wheel/bdist_wheel.py`. When I signed off on the change in packaging, I actually expected pip to upgrade their vendored version quicker. Right now GraalPy is in this in-between state there, with pip having the old vendored packaging, wheel having a copy the logic in [bdist_wheel.py](https://github.com/pypa/wheel/blob/main/src/wheel/bdist_wheel.py#L91), and packages like maturin and meson-python already implementing the new logic. I hope it'll just slowly resolve as the necessary changes that started with that packaging commit slowly perculate through everywhere.

Reply:
xref https://github.com/python/cpython/issues/99560

Reply:
Here's the wheel PR: https://github.com/pypa/wheel/pull/557

Reply:
marking as resolved in this PR. follow up in #1520